federation/server_room.go

```diff
@@ -498,21 +498,27 @@ func (i *ServerRoomImplDefault) GenerateSendJoinResponse(room *ServerRoom, s *Se
 	authEvents := room.AuthChainForEvents(stateEvents)
 
 	// get servers in room *before* the join event
-	serversInRoom := []string{s.serverName}
+	serversInRoom := []spec.ServerName{s.serverName}
 	if !omitServersInRoom {
 		serversInRoom = room.ServersInRoom()
 	}
 
+	serversInRoomStrings := make([]string, len(serversInRoom))
+	for i, serverName := range serversInRoom {
+		serversInRoomStrings[i] = string(serverName)
+	}
+
 	// insert the join event into the room state
 	room.AddEvent(joinEvent)
 	log.Printf("Received send-join of event %s", joinEvent.EventID())
 
 	// return state and auth chain
 	return fclient.RespSendJoin{
-		Origin:         spec.ServerName(s.serverName),
+		Origin:         s.serverName,
 		AuthEvents:     gomatrixserverlib.NewEventJSONsFromEvents(authEvents),
 		StateEvents:    gomatrixserverlib.NewEventJSONsFromEvents(stateEvents),
 		MembersOmitted: expectPartialState,
-		ServersInRoom:  serversInRoom,
+		// TODO: It feels like `ServersInRoom` should be `[]spec.ServerName` instead of `[]string`
+		ServersInRoom: serversInRoomStrings,
```

Comment:
This feels like a [`gomatrixserverlib` problem](https://github.com/matrix-org/gomatrixserverlib/blob/479c9abf1ca5d5c256ebea5b1d8f6a5c7b69f94d/fclient/federationtypes.go#L279-L280).

(not going to fix in this PR)